packages/sdk/src/index.ts

```diff
@@ -46,8 +46,11 @@ export type SerializableEvent = Omit<
 
 // this convoluted type is meant to cast events to the types you need based on the contract and event name
 // example: type NewContractRegistered = GetEventType<Registry,"NewContractRegistered">;
+// TODO: the any below is a hacky solution because some typechain types fail to resolve due to
+// incompatible TypedEventFilter and TypedEvent types. This will be fixed by upgrading typechain
+// to a version where Ethers events are exported as first class types.
 export type GetEventType<ContractType extends Contract, EventName extends string> = ReturnType<
   ContractType["filters"][EventName] extends Callable ? ContractType["filters"][EventName] : never
 > extends TypedEventFilter<infer T, infer S>
-  ? TypedEvent<T & S extends Result ? T & S : never>
+  ? TypedEvent<T & S extends Result ? T & S : any>
```

Comment:
i don tthink this change is that big of a deal, would be nice to totally get rid of this with new typechain